azure-mgmt-graph-rbac/src/main/java/com/microsoft/azure/management/graphrbac/ActiveDirectoryUser.java

```diff
@@ -8,40 +8,217 @@
-import com.microsoft.azure.management.resources.fluentcore.arm.models.HasId;
-import com.microsoft.azure.management.resources.fluentcore.arm.models.HasManager;
-import com.microsoft.azure.management.resources.fluentcore.arm.models.HasName;
+import com.microsoft.azure.management.resources.fluentcore.arm.CountryIsoCode;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
 import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import com.microsoft.azure.management.resources.fluentcore.model.Updatable;
 
 /**
  * An immutable client-side representation of an Azure AD user.
  */
 @Fluent(ContainerName = "/Microsoft.Azure.Management.Graph.RBAC.Fluent")
 @Beta
 public interface ActiveDirectoryUser extends
+        ActiveDirectoryObject,
         HasInner<UserInner>,
-        HasId,
-        HasName,
-        HasManager<GraphRbacManager> {
+        Updatable<ActiveDirectoryUser.Update> {
     /**
-     * @return Gets or sets user principal name.
+     * @return Gets or sets user principal name
      */
     String userPrincipalName();
 
     /**
-     * @return Gets or sets user signIn name.
+     * @return Gets or sets user signIn name
      */
     String signInName();
 
     /**
-     * @return Gets or sets user mail.
+     * @return Gets or sets user mail
      */
     String mail();
 
     /**
-     * @return The mail alias for the user.
+     * @return The mail alias for the user
      */
     String mailNickname();
+
+    /**
+     * @return if the usage location of the uer
+     */
+    CountryIsoCode usageLocation();
+
+    /**************************************************************
+     * Fluent interfaces to provision a user
+     **************************************************************/
+
+    /**
+     * Container interface for all the definitions that need to be implemented.
+     */
+    interface Definition extends
+            DefinitionStages.Blank,
+            DefinitionStages.WithUserPrincipalName,
+            DefinitionStages.WithPassword,
+            DefinitionStages.WithCreate {
+    }
+
+    /**
+     * Grouping of all the user definition stages.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of the user definition.
+         */
+        interface Blank extends WithUserPrincipalName {
+        }
+
+        /**
+         * A user definition allowing user principal name to be specified.
+         */
+        interface WithUserPrincipalName {
+            /**
+             * Specifies the user principal name of the user. It must contain one of
+             * the verified domains for the tenant.
+             * @param userPrincipalName the user principal name
+             * @return the next stage of user definition
+             */
+            WithPassword withUserPrincipalName(String userPrincipalName);
```

Comment:
if the `ActiveDirectoryUser` type shipped in a previous version, but these withers are newly introduced in v1.2, then then should individually have `@Beta(SinceVersion.V1_2_0)` on them